test/util/e2e.go

```diff
@@ -232,3 +231,26 @@ func RestartKueueController(ctx context.Context, k8sClient client.Client) {
 	kcmKey := types.NamespacedName{Namespace: "kueue-system", Name: "kueue-controller-manager"}
 	rolloutOperatorDeployment(ctx, k8sClient, kcmKey)
 }
+
+func WaitForActivePodsAndTerminate(ctx context.Context, k8sClient client.Client, restClient *rest.RESTClient, cfg *rest.Config, namespace string, activePodsCount, exitCode int) {
+	pods := corev1.PodList{}
+	gomega.Eventually(func(g gomega.Gomega) bool {
+		g.Expect(k8sClient.List(ctx, &pods, client.InNamespace(namespace))).To(gomega.Succeed())
+		g.Expect(len(pods.Items)).To(gomega.BeEquivalentTo(activePodsCount))
+		for _, p := range pods.Items {
+			if (len(p.Status.PodIP) == 0 || p.Status.PodIP == "0.0.0.0") || (p.Status.Phase != corev1.PodRunning && p.Status.Phase != corev1.PodPending) {
+				return false
+			}
+		}
+		return true
```

Comment:
Please do the filtering to compute `activePods` list. Then check `g.Expect(activePods).To(gomega.HaveLen(activePodsCount))`. This will produce more informative failure messages when it fails.